clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -181,33 +307,469 @@ public void updateState(ConsumerGroupHeartbeatResponseData response) {
     public void transitionToFenced() {
         resetEpoch();
         transitionTo(MemberState.FENCED);
+
+        // Release assignment
+        CompletableFuture<Void> callbackResult = invokeOnPartitionsRevokedOrLostToReleaseAssignment();
```

Comment:
As previously mentioned, it is clearer to directly invoke `invokeOnPartitionsLost` here. Is there a case we don't invoke onPartitionsLost?

Reply:
Agree, done here. Note my answer above though, about the cases where we do need revoke or lost, that one should stay.